torch/testing/_internal/common_utils.py

```diff
@@ -829,8 +829,13 @@ def assertTensorsEqual(a, b):
                                 self.assertTrue(torch.equal(inf_sign, torch.isinf(b).sign()), message)
                                 diff[inf_mask] = 0
                         # TODO: implement abs on CharTensor (int8)
-                        if diff.is_signed() and diff.dtype != torch.int8:
+                        # TODO: modify abs to return float/double for ComplexFloat/ComplexDouble
+                        if diff.dtype.is_complex or (diff.is_signed() and diff.dtype != torch.int8):
                             diff = diff.abs()
+                            if diff.dtype == torch.complex64:
+                                diff = diff.to(torch.float)
```

Comment:
Errr... does this truncate the complex portion, or compute absolute value? If it truncates, you'll fail to catch differences in the imaginary part...

Reply:
diff contains the magnitude of the complex numbers and so the imaginary part is expected to be zero. The reason why we need to use to right now is that abs() when called on a complex tensor returns a complex tensor instead of float/double tensor

Reply:
Might be worth a little note, if you think so.